src/beast/beast/module/core/files/File.cpp

```diff
@@ -28,23 +28,11 @@
 
 namespace beast {
 
-// We need to make a shared singleton or else there are
-// issues with the leak detector and order of detruction.
-//
-class NonexistentHolder
-{
-public:
-    static NonexistentHolder* getInstance()
-    {
-        return SharedSingleton <NonexistentHolder>::getInstance();
-    }
-
-    File file;
-};
+static File s_nonexistent;
 
-File const& File::nonexistent ()
+File const& File::nonexistent()
 {
-    return NonexistentHolder::getInstance ()->file;
+    return s_nonexistent;
```

Comment:
@HowardHinnant Putting the static in the function is my usual practice - it works around order of initialization issues. But in this case it might be simpler to keep it outside. It'll be initialized before we have to worry about threads (something we can't say if we put it in the function). So if we don't have order of initialization issues (I'll have to check if we might) then we might want to keep it outside.